sync/awsdatabricks.py

```diff
@@ -273,17 +277,17 @@ def missing_message(input: str) -> str:
             + "https://docs.synccomputing.com/sync-gradient/integrating-with-gradient/databricks-workflows"
         )
 
-    if not cluster_info or not cluster_info["Reservations"]:
+    if not cluster_info or not cluster_info.get("Reservations"):
         reservations_response = Response(
             error=DatabricksError(message=missing_message("instances"))
         )
     else:
         reservations_response = Response(result={"Reservations": cluster_info["Reservations"]})
 
-    if not cluster_info or not cluster_info["Volumes"]:
+    if not cluster_info:
         volumes_response = Response(error=DatabricksError(message=missing_message("ebs volumes")))
     else:
-        volumes_response = Response(result={"Volumes": cluster_info["Volumes"]})
+        volumes_response = Response(result={"Volumes": cluster_info.get("Volumes", [])})
```

Comment:
Just above we're returning an error for missing volume information. Volume information is potentially missing here as well - shouldn't these 2 clauses be consistent?

Reply:
For the current product getting the EBS information isn't critical, so I decided to not have it throw an error if ONLY the ebs volume data is missing. If the cluster info is missing entirely then an error object still needs to be returned for both EBS & Reservations, which is why the logic is the way it is nowl

Reply:
Ok. For instances return an error whenever we have no instance info to return. For volumes return an error only when both volume and instance info are missing? The volume part of the tuple response seems to be representing more than just the state of volume info. I think it's easier to wrap your head around the behavior of this function if each response object is only concerned with the data that it'd contain in a successful response. I understand volume info isn't as critical as instance info - would returning an error for missing volume info break anything? This method is not part of the developer interface however, so this isn't going to hold up the PR.

Reply:
Good point, I can massage this a little bit more to make it more clear.

Reply:
Yeah that's a good call. I moved the error handling up to the `_get_cluster_report`, where it makes a lot more sense to be.